delivery_carrier_label_paazl/models/stock_picking.py

```diff
@@ -0,0 +1,239 @@
+            ),
+        )
+
+    def _paazl_quant_to_order_data(self, quant):
+        """Return a dict describing a quant for the `order` endpoint"""
+        result = {
+            "quantity": int(quant.quantity),
+            "weight": quant.product_id.weight,
+            "volume": quant.product_id.volume,
+            "code": quant.product_id.default_code,
+            "description": quant.product_id.description_pickingout or "/",
+            "hsTariffCode": quant.product_id.hs_code,
+        }
+        # support product_dimension if installed
+        for name in ["length", "height", "width"]:
+            field_name = "product_%s" % name
+            if field_name not in quant.product_id._fields:
+                continue
+            result[name] = quant.product_id[field_name]
+            uom = quant.product_id.dimensional_uom_id
+            if uom:
+                result[name] = uom._compute_quantity(
+                    result[name], self.env.ref("uom.product_uom_cm"),
+                )
+            result[name] = int(result[name])
+        return result
+
+    def _paazl_commit_order_data(self):
+        """Return a dict that can be passed as parameters to the `commitOrder`
+        endpoint of the paazl soap api"""
+        self.ensure_one()
+        shipping_address = self._paazl_partner_to_order_data(self.partner_id)
+        shipping_address["customerName"] = shipping_address.pop("addresseeLine")
+        if self.partner_id.parent_id:
+            shipping_address["companyName"] = self.partner_id.parent_id.name
+        return dict(
+            **self._paazl_auth_order_id(),
+            pendingOrderReference=self._paazl_reference(),
+            customerEmail=self.partner_id.email,
+            customerPhoneNumber=self.partner_id.phone,
+            shippingMethod=dict(
+                type="delivery",
+                identifier=0,
+                option=self.option_ids[:1].code,
+                orderWeight=sum(
+                    product["quantity"] * product["weight"]
+                    for product in self._paazl_order_data()["products"]["product"]
+                ),
+                description=self.note or "/",
+                packageCount=len(self.package_ids),
+            ),
+            shipperAddress=self._paazl_partner_to_order_data(
+                self.picking_type_id.warehouse_id.partner_id,
+            ),
+            shippingAddress=shipping_address,
+            totalAmount=0,
+        )
+
+    def _paazl_partner_to_order_data(self, partner):
+        """Return a dict describing a partner for the `commitOrder` endpoint"""
+        result = {
+            "addresseeLine": partner.name,
+            "street": partner.street,
+            "housenumber": "",
+            "zipcode": partner.zip,
+            "city": partner.city,
+            "province": partner.state_id.code,
+            "country": partner.country_id.code,
+        }
+        # support base_address_extended if installed
```

Comment:
I was told paazl does not accept addresses without a house number at all. Maybe base_address_extended has to be a hard dependency of this module?

Reply:
in my tests it happily accepted the empty string as house number, so it seems to me if the empty string breaks things, that's carrier specific. That plus the fact that `base_address_extended` has some nasty side effects if not installed with care ([the street format issue](https://github.com/OCA/OpenUpgrade/pull/2408)) makes me prefer not to set a dependency

Reply:
OK!